### PR TITLE
Api V2

### PR DESCRIPTION
**Pull request by gatzjames**
Feature branch for Insomnia API v2.
In progress.

TODO:
- [x] Point INSOMNIA_API_URL and INSOMNIA_WEBSITE_URL variables to production instead of staging.
- [x] Remove staging url forms in settings and login page, and enabled ai check
- [x] Test edge cases for data migrations.
- [x] Update E2E tests for ScratchPad and add tests for the Logged In user flow
- [x] UX for login code missing or incorrect
- [x] UX for migration failed
- [x] examine migration with updates rather than creates
- [x] examine migration as one by one rather than batch stages
- [x] UX for errors in migration flow

### CODE REVIEW

File: packages/insomnia/src/main/insomniaFetch.ts [hunk 1]
```diff
@@ -30,6 +31,8 @@ const exponentialBackOff = async (url: string, init: RequestInit, retries = 0):
       return exponentialBackOff(url, init, retries);
     }
     if (!response.ok) {
+      // TODO: review error status code behaviour with backend, should we parse errors here and return response
+      // or should we rethrow an error with a response object inside? should we be exposing errors to the app UI?
       console.log(`Response not OK: ${response.status} for ${url}`);
```
Comment (jackkav) [inline packages/insomnia/src/main/insomniaFetch.ts:36]: ideally we can offer a non-blocking ux in which 5xx errors are treated an temporary downtime and 4xx errors have a planned ux based on status code? one example could be a 5 second flash of text in the status bar indicating a failed fetch and reason.

File: packages/insomnia/src/models/settings.ts [hunk 2]
```diff
@@ -117,6 +117,12 @@ export async function getOrCreate() {
   return results[0];
 }
 
+export async function get() {
+  const results = await db.all<Settings>(type) || [];
+
+  return results[0];
+}
+
```
Comment (jackkav) [inline packages/insomnia/src/models/settings.ts:125]: the need for this addition is not clear? am I missing something?

File: packages/insomnia/src/models/environment.ts [hunk 3]
```diff
@@ -70,13 +70,26 @@ export async function getOrCreateForParentId(parentId: string) {
...
+      return baseEnvironment;
+    } catch (e) {
+      const existingEnvironment = await getById(baseEnvironmentId);
+
+      if (existingEnvironment) {
+        return existingEnvironment;
+      }
+
+      throw e;
+    }
```
Comment (jackkav) [inline packages/insomnia/src/models/environment.ts:92]: this needs attention, very spooky

File: packages/insomnia/src/models/project.ts [hunk 4]
```diff
@@ -77,3 +78,25 @@ export async function all() {
...
+    }
+
+    if (!scratchPad) {
+      console.log('Initializing Scratch Pad');
+      await workspace.create({ _id: 'wrk_scratchpad', name: 'Scratch Pad', parentId: DEFAULT_PROJECT_ID, scope: 'collection' });
+    }
+  } catch (err) {
+    console.warn('Failed to create default project. It probably already exists', err);
+  }
+}
```
Comment (jackkav) [inline packages/insomnia/src/models/project.ts]: I would advise moving this code to the specific place it is intended to run and dealing with the smells.

File: packages/insomnia/src/sync/vcs/migrate-to-cloud-projects.ts [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+          }
+        } catch (e) {
+          console.warn('Failed to initialize sync on workspace. This will be retried when the workspace is opened on the app.', e);
+        }
+      }
+    }
+
+    status = 'completed';
+  } catch (err) {
+    console.warn('Failed to migrate projects to cloud', err);
```
Comment (jackkav) [inline packages/insomnia/src/sync/vcs/migrate-to-cloud-projects.ts:113]: needs a ux

File: packages/insomnia/src/ui/components/avatar.tsx [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+  constructor({ ttl }: { ttl: number }) {
+    this.ttl = ttl;
+  }
+
+  read(src: string) {
+    const now = Date.now();
+    if (this.__cache[src] && typeof this.__cache[src].value !== 'string') {
+      // If the value is a Promise, throw it to indicate that the cache is still loading
+      throw this.__cache[src].value;
+    } else if (this.__cache[src] && now - this.__cache[src].timestamp < this.ttl) {
```
Comment (jackkav) [inline packages/insomnia/src/ui/components/avatar.tsx:103]: curious inconsistency here, why choose a class for this caching behaviour we typically use maps for?

File: packages/insomnia/src/ui/components/dropdowns/request-actions-dropdown.tsx [hunk 7]
```diff
@@ -19,15 +19,14 @@ import type { RequestAction } from '../../../plugins';
 import { getRequestActions } from '../../../plugins';
 import * as pluginContexts from '../../../plugins/context/index';
 import { useRequestMetaPatcher, useRequestPatcher } from '../../hooks/use-request';
-import { RootLoaderData } from '../../routes/root';
-import { type DropdownProps } from '../base/dropdown';
+import { useRootLoaderData } from '../../routes/root';
```
Comment (jackkav) [inline packages/insomnia/src/ui/components/dropdowns/request-actions-dropdown.tsx:22]: This feels bad only for me because in british english, root and route sound the same

File: packages/insomnia/src/ui/components/organization-avatar.tsx [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+const Avatar = ({ src, alt }: { src: string; alt: string }) => {
+  imgCache.read(src);
+  return <img
+    src={src}
+    alt={alt}
+    className='h-full w-full aspect-square object-cover'
+    aria-label={alt}
+  />;
+};
+
```
Comment (jackkav) [inline packages/insomnia/src/ui/components/organization-avatar.tsx:70]: above seems like copypasta from avatar.tsx

File: packages/insomnia/src/ui/context/app/presence-context.tsx [hunk 9]
```diff
@@ -0,0 +1,194 @@
+import React, { createContext, FC, PropsWithChildren, useContext, useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+import { useInterval } from 'react-use';
+
+import { getCurrentSessionId } from '../../../account/session';
+
+const PresenceContext = createContext<{
+  presence: UserPresence[];
+}>({
+  presence: [],
```
Comment (jackkav) [inline packages/insomnia/src/ui/context/app/presence-context.tsx:11]: in order to simplify updates and get free de-duping, can this state be modelled as an object rather than an array?

File: packages/insomnia/src/ui/context/app/presence-context.tsx [hunk 10]
```diff
@@ -0,0 +1,194 @@
...
+    workspaceId,
+  } = useParams() as {
+    organizationId: string;
+    projectId: string;
+    workspaceId: string;
+  };
+
+  const [presence, setPresence] = useState<UserPresence[]>([]);
+
+  // Update presence when the app window closes
```
Comment (jackkav) [inline packages/insomnia/src/ui/context/app/presence-context.tsx]: why on window close? perhaps update this comment to show the significance of leaving a session

File: packages/insomnia/src/ui/index.tsx [hunk 11]
```diff
@@ -42,24 +48,91 @@ initializeLogging();
...
+      return '/organization';
+    }
+
+    if (hasUserLoggedInBefore) {
+      return '/auth/login';
+    }
+
+    return '/scratchpad';
+  } catch (e) {
+    return '/organization';
```
Comment (jackkav) [inline packages/insomnia/src/ui/index.tsx:99]: this fallback makes org redirect harder to reason about because it could be either logged in or out.

File: packages/insomnia/src/ui/index.tsx [hunk 12]
```diff
@@ -42,24 +48,91 @@ initializeLogging();
...
       children: [
+        {
+          path: 'dev',
+          action: async (...args) => (await import('./routes/dev')).action(...args),
+          element: <Dev />,
+        },
+        {
+          path: '/scratchpad',
+          loader: async (...args) => (await import('./routes/scratchpad')).loader(...args),
+        },
```
Comment (jackkav) [inline packages/insomnia/src/ui/index.tsx:123]: does this need its own route? especially given that it is currently modelled as hardcoded proj id and wrk id. It would also need the debug view to detect this hardcoded workspace id in order to add the scratch pad UI flair.

File: packages/insomnia/src/ui/index.tsx [hunk 13]
```diff
@@ -802,7 +902,7 @@ router.subscribe(({ location }) => {
 
 async function renderApp() {
   await database.initClient();
-
+  await models.project.seed();
```
Comment (jackkav) [inline packages/insomnia/src/ui/index.tsx]: smelly

File: packages/insomnia/src/models/settings.ts (hunk 2 above)
Comment (jackkav) [reply packages/insomnia/src/models/settings.ts:125]: This change is to avoid a on startup race condition that can occasionally cause two create function to fire at once.

File: packages/insomnia/src/ui/components/avatar.tsx (hunk 6 above)
Comment (jackkav) [reply packages/insomnia/src/ui/components/avatar.tsx:103]: If this is intended just to be a temporary copypasta perhaps a comment explaining and pointing at where it came from would help.

File: packages/insomnia/src/ui/context/app/presence-context.tsx [hunk 14]
```diff
@@ -0,0 +1,194 @@
...
+              data: {
+                project: projectId,
+                file: workspaceId,
+              },
+            });
+
+          const { data } = response;
+          if (data.length > 0) {
+            setPresence(response.data);
+          }
```
Comment (jackkav) [inline packages/insomnia/src/ui/context/app/presence-context.tsx]: this is handled inconsistently with the function above above

File: packages/insomnia-smoke-test/playwright.config.ts [hunk 15]
```diff
@@ -20,6 +20,12 @@ const config: PlaywrightTestConfig = {
       testMatch: /critical\/.*.test.ts/,
       retries: 0,
     },
+    {
+      // Single critical path test, runs on release recurring
+      name: 'Migration',
+      testMatch: /migration\/.*.test.ts/,
+      retries: 0,
```
Comment (jackkav) [inline packages/insomnia-smoke-test/playwright.config.ts:27]: move into smoke folder

File: packages/insomnia-smoke-test/tests/migration/local-to-cloud-projects.test.ts [hunk 16]
```diff
@@ -0,0 +1,74 @@
...
+  },
+  userConfig: async ({ userConfig }, use) => {
+    await use({
+      ...userConfig,
+      session: undefined,
+    });
+  },
+});
+
+testWithLegacyDatabase('Run data migration to version 8', async ({ page, userConfig }) => {
```
Comment (jackkav) [inline packages/insomnia-smoke-test/tests/migration/local-to-cloud-projects.test.ts:19]: tests that local + legacy remote projects, will have their parentIds set to org id, and new cloud projects works

File: packages/insomnia/package.json [hunk 17]
```diff
@@ -58,6 +58,7 @@
     "electron-context-menu": "^3.6.1",
     "electron-log": "^4.4.8",
     "grpc-reflection-js": "0.3.0",
+    "gsap": "^3.11.3",
```
Comment (jackkav) [inline packages/insomnia/package.json:61]: for the animated svg background

File: packages/insomnia/src/common/export-all-data.ts [hunk 18]
```diff
@@ -0,0 +1,359 @@
...
+  [unitTest.type]: EXPORT_TYPE_UNIT_TEST,
+  [workspace.type]: EXPORT_TYPE_WORKSPACE,
+  [cookieJar.type]: EXPORT_TYPE_COOKIE_JAR,
+  [environment.type]: EXPORT_TYPE_ENVIRONMENT,
+  [apiSpec.type]: EXPORT_TYPE_API_SPEC,
+  [protoFile.type]: EXPORT_TYPE_PROTO_FILE,
+  [protoDirectory.type]: EXPORT_TYPE_PROTO_DIRECTORY,
+}[type] || undefined);
+
+export async function exportWorkspaceData({
```
Comment (jackkav) [inline packages/insomnia/src/common/export-all-data.ts:40]: TODO: add workspace alongside this

File: packages/insomnia/src/main/insomniaFetch.ts [hunk 19]
```diff
@@ -30,6 +31,8 @@ const exponentialBackOff = async (url: string, init: RequestInit, retries = 0):
       return exponentialBackOff(url, init, retries);
```
Comment (jackkav) [inline packages/insomnia/src/main/insomniaFetch.ts:31]: limit retries

File: packages/insomnia/src/sync/vcs/initialize-model-from.ts [hunk 20]
```diff
@@ -10,6 +10,7 @@ export const initializeProjectFromTeam = (team: Team) => models.initModel<Remote
   {
     _id: `${models.project.prefix}_${team.id}`,
     remoteId: team.id,
+    parentId: team.id,
```
Comment (jackkav) [inline packages/insomnia/src/sync/vcs/initialize-model-from.ts]: should be team.organizationId

File: packages/insomnia/src/sync/vcs/vcs.ts [hunk 21]
```diff
@@ -1126,24 +1135,27 @@ export class VCS {
...
 
     if (!teamId || !teamPublicKeys?.length) {
       throw new Error('teamId and teamPublicKeys must not be null or empty!');
     }
 
     // Encrypt the symmetric key with the public keys of all the team members, ourselves included
-    for (const { accountId, publicKey } of teamPublicKeys) {
+    for (const { accountId, publicKey, autoLinked } of teamPublicKeys) {
       teamKeys.push({
+        autoLinked,
```
Comment (jackkav) [inline packages/insomnia/src/sync/vcs/vcs.ts:1158]: this is going to cause a bug someday

File: packages/insomnia/src/ui/components/avatar.tsx [hunk 22]
```diff
@@ -0,0 +1,184 @@
...
+        };
+        img.src = src;
+      });
+      this.__cache[src] = { value: promise, timestamp: now };
+      throw promise;
+    }
+  }
+}
+
+// Cache images for 10 minutes
```
Comment (jackkav) [inline packages/insomnia/src/ui/components/avatar.tsx:123]: could be solved with cache headers on the image, rather than a custom cache implementation here and in org avatar.tsx

File: packages/insomnia/src/ui/components/dropdowns/sync-dropdown.tsx [hunk 23]
```diff
@@ -348,6 +366,35 @@ export const SyncDropdown: FC<Props> = ({ vcs, workspace, project }) => {
     },
   }];
 
+  const isGitSyncEnabled = features.gitSync.enabled;
```
Comment (jackkav) [inline packages/insomnia/src/ui/components/dropdowns/sync-dropdown.tsx:367]: add some fallback here

File: packages/insomnia/src/ui/components/modals/settings-modal.tsx [hunk 24]
```diff
@@ -87,6 +132,9 @@ export const SettingsModal = forwardRef<SettingsModalHandle, ModalProps>((props,
               <AI />
             </PanelContainer>
           </TabItem>
+          <TabItem key="dev" title="Dev">
+            <Dev />
+          </TabItem>
```
Comment (jackkav) [inline packages/insomnia/src/ui/components/modals/settings-modal.tsx]: remove this

File: packages/insomnia/src/ui/components/panes/project-empty-state-pane.tsx [hunk 25]
```diff
@@ -52,51 +108,143 @@ interface Props {
 }
 
 export const EmptyStatePane: FC<Props> = ({ createRequestCollection, createDesignDocument, importFrom, cloneFromGit }) => {
+  const { organizationId } = useParams<{ organizationId: string }>();
+  const { organizations } = useOrganizationLoaderData();
+  const currentOrg = organizations.find(organization => (organization.id === organizationId));
+  const { features } = useRouteLoaderData(':organizationId') as { features: FeatureList };
+
+  const isGitSyncEnabled = features.gitSync.enabled;
```
Comment (jackkav) [inline packages/insomnia/src/ui/components/panes/project-empty-state-pane.tsx:116]: needs a fallback

Comment (jackkav) [reply packages/insomnia/src/ui/components/panes/project-empty-state-pane.tsx:116]: needs to not be copypasta in three places

File: packages/insomnia/src/ui/components/present-users.tsx [hunk 26]
```diff
@@ -0,0 +1,44 @@
...
+
+  if (!presence) {
+    return null;
+  }
+
+  const accountId = getAccountId();
+
+  const activeUsers = presence
+    .filter(p => {
+      return p.project === projectId && p.file === workspaceId;
```
Comment (jackkav) [inline packages/insomnia/src/ui/components/present-users.tsx]: change to remoteId

File: packages/insomnia/src/ui/context/app/ai-context.tsx [hunk 27]
```diff
@@ -39,26 +38,26 @@ export const AIProvider: FC<PropsWithChildren> = ({ children }) => {
...
         method: 'post',
-        action: `/organization/${organizationId}/project/${projectId}/workspace/${workspaceId}/ai/access`,
+        action: `/organization/${organizationId}/ai/access`,
       });
     }
-  }, [aiAccessFetcher, organizationId, projectId, workspaceId, loggedIn, prevProjectId]);
+  }, [aiAccessFetcher, organizationId, loggedIn]);
 
-  const isAIEnabled = aiAccessFetcher.data?.enabled ?? false;
+  // const isAIEnabled = Boolean(aiAccessFetcher.data?.enabled);
```
Comment (jackkav) [inline packages/insomnia/src/ui/context/app/ai-context.tsx]: uncomment this, and line 108

File: packages/insomnia/src/ui/context/app/presence-context.tsx [hunk 28]
```diff
@@ -0,0 +1,191 @@
...
+
+        const rows = response?.data || [];
+        if (rows.length > 0) {
+          setPresence(rows);
+        }
+      } catch (e) {
+        console.log('Error parsing response', e);
+      }
+    }
+  }, 1000 * 30);
```
Comment (jackkav) [inline packages/insomnia/src/ui/context/app/presence-context.tsx]: change to 60 seconds

File: packages/insomnia/src/ui/context/app/presence-context.tsx [hunk 29]
```diff
@@ -0,0 +1,191 @@
...
+      } catch (e) {
+        console.log('Error parsing response', e);
+      }
+    };
+
+    window.addEventListener('beforeunload', handleWindowClose);
+    return () => {
+      window.removeEventListener('beforeunload', handleWindowClose);
+    };
+  }, [organizationId]);
```
Comment (jackkav) [inline packages/insomnia/src/ui/context/app/presence-context.tsx]: remove this

File: packages/insomnia/src/ui/hooks/use-global-keyboard-shortcuts.ts [hunk 30]
```diff
@@ -4,14 +4,14 @@ import * as plugins from '../../plugins';
...
 import { useSettingsPatcher, useWorkspaceMetaPatcher } from './use-request';
 export const useGlobalKeyboardShortcuts = () => {
   const workspaceData = useRouteLoaderData(':workspaceId') as WorkspaceLoaderData | undefined;
   const {
     settings,
-  } = useRouteLoaderData('root') as RootLoaderData;
+  } = useRootLoaderData();
   const { activeWorkspaceMeta } = workspaceData || {};
   const patchSettings = useSettingsPatcher();
   const patchWorkspaceMeta = useWorkspaceMetaPatcher();
```
Comment (jackkav) [inline packages/insomnia/src/ui/hooks/use-global-keyboard-shortcuts.ts:17]: keyboard shortcuts should not work when logged out

File: packages/insomnia/src/ui/index.tsx [hunk 31]
```diff
@@ -42,24 +51,90 @@ initializeLogging();
 document.body.setAttribute('data-platform', process.platform);
 document.title = getProductName();
 
-let locationHistoryEntry = `/organization/${DEFAULT_ORGANIZATION_ID}/project/${DEFAULT_PROJECT_ID}`;
-const prevLocationHistoryEntry = localStorage.getItem('locationHistoryEntry');
+try {
+  const skipOnboarding = getSkipOnboarding();
```
Comment (jackkav) [inline packages/insomnia/src/ui/index.tsx:56]:
```suggestion
// This allows us to inject state from env vars in order to run playwight tests that simulate a logged in user
  const skipOnboarding = getSkipOnboarding();
```

File: packages/insomnia/src/ui/index.tsx [hunk 32]
```diff
@@ -42,24 +51,90 @@ initializeLogging();
...
+      return '/organization';
+    }
+
+    if (hasUserLoggedInBefore) {
+      return '/auth/login';
+    }
+
+    return '/scratchpad';
+  } catch (e) {
+    return '/organization';
```
Comment (jackkav) [inline packages/insomnia/src/ui/index.tsx]:
```suggestion
    return '/scratchpad';
```

File: packages/insomnia/src/ui/routes/actions.tsx [hunk 33]
```diff
@@ -32,9 +30,44 @@ export const createNewProjectAction: ActionFunction = async ({ request, params }
...
+      _id: newCloudProject.id,
+      name: newCloudProject.name,
+      remoteId: newCloudProject.id,
+      parentId: organizationId,
+    });
+
+    return redirect(`/organization/${organizationId}/project/${project._id}`);
+  } catch (err) {
+    console.log(err);
+    return null;
```
Comment (jackkav) [inline packages/insomnia/src/ui/routes/actions.tsx]:
```suggestion
    return { error: err.message };
```

File: packages/insomnia/src/ui/routes/actions.tsx [hunk 34]
```diff
@@ -53,14 +86,34 @@ export const renameProjectAction: ActionFunction = async ({
...
+        error: response.error === 'FORBIDDEN' ? 'You do not have permission to rename this project.' : 'An unexpected error occurred while renaming the project. Please try again.',
+      };
+    }
 
-  return null;
+    await models.project.update(project, { name });
+    return null;
+  } catch (err) {
+    console.log(err);
+    return null;
```
Comment (jackkav) [inline packages/insomnia/src/ui/routes/actions.tsx]:
```suggestion
    return { error: err.message };
```

File: packages/insomnia/src/ui/routes/actions.tsx [hunk 35]
```diff
@@ -70,12 +123,33 @@ export const deleteProjectAction: ActionFunction = async ({ params }) => {
...
 
-  window.main.trackSegmentEvent({ event: SegmentEvent.projectLocalDelete });
+    await models.stats.incrementDeletedRequestsForDescendents(project);
+    await models.project.remove(project);
 
-  return redirect(`/organization/${DEFAULT_ORGANIZATION_ID}/project/${DEFAULT_PROJECT_ID}`);
+    return redirect(`/organization/${organizationId}`);
+  } catch (err) {
+    console.log(err);
+    return null;
```
Comment (jackkav) [inline packages/insomnia/src/ui/routes/actions.tsx]:
```suggestion
    return { error: err.message };
```

File: packages/insomnia/src/ui/routes/organization.tsx [hunk 36]
```diff
@@ -1,56 +1,629 @@
...
+              isOwnerOfOrganization({
+                organization: b,
+                accountId: accountId || '',
+              })
+            ) {
+              return 1;
+            } else {
+              return 0;
+            }
+          }),
```
Comment (jackkav) [inline packages/insomnia/src/ui/routes/organization.tsx]: use destructure instead of sort here

File: packages/insomnia/src/ui/routes/organization.tsx [hunk 37]
```diff
@@ -1,56 +1,629 @@
...
+export interface FeatureList {
+  gitSync: FeatureStatus;
+  orgBasicRbac: FeatureStatus;
+}
+
+export const singleOrgLoader: LoaderFunction = async ({ params }) => {
+  const { organizationId } = params;
+
+  try {
+    const response = await window.main.insomniaFetch<{ features: FeatureList }>({
```
Comment (jackkav) [inline packages/insomnia/src/ui/routes/organization.tsx]:
```suggestion
    const response = await window.main.insomniaFetch<{ features: FeatureList } | void>({
```

File: packages/insomnia/src/ui/routes/project.tsx [hunk 38]
```diff
@@ -40,31 +43,60 @@ import { ApiSpec } from '../../models/api-spec';
...
+}
+
+async function getAllTeamProjects(organizationId: string) {
+  const sessionId = getCurrentSessionId() || '';
+  console.log('Fetching projects for team', organizationId);
+  if (!sessionId) {
+    return [];
+  }
+
+  const response = await window.main.insomniaFetch<{
```
Comment (jackkav) [inline packages/insomnia/src/ui/routes/project.tsx:86]: review the insomnia fetches to return empty arrays/objects when response is 4xx/5xx?

File: packages/insomnia/src/ui/routes/project.tsx [hunk 39]
```diff
@@ -323,8 +412,47 @@ const ProjectRoute: FC = () => {
...
+      });
+    return {
+      ...workspace,
+      presence: workspacePresence,
+    };
+  });
+
+  const projectsWithPresence = projects.map(project => {
+    const projectPresence = presence
+      .filter(p => p.project === project._id)
```
Comment (jackkav) [inline packages/insomnia/src/ui/routes/project.tsx]: use remoteId

File: packages/insomnia/src/ui/routes/project.tsx [hunk 40]
```diff
@@ -382,8 +511,48 @@ const ProjectRoute: FC = () => {
...
+        onDone: async (isYes: boolean) => {
+          if (isYes) {
+            window.main.openInBrowser(`${getAppWebsiteBaseURL()}/app/subscription/update?plan=team`);
+          }
+        },
+      }) : showModal(AlertModal, {
+        title: 'Upgrade Plan',
+        message: 'Git Sync is only enabled for Team plan or above, please ask the organization owner to upgrade.',
+      });
+  };
```
Comment (jackkav) [inline packages/insomnia/src/ui/routes/project.tsx:590]: the third copypasta

File: packages/insomnia/src/ui/routes/workspace.tsx [hunk 41]
```diff
@@ -110,10 +110,7 @@ export const workspaceLoader: LoaderFunction = async ({
 
   const allProjects = await models.project.all();
 
-  const organizationProjects =
-    organizationId === DEFAULT_ORGANIZATION_ID
-      ? allProjects.filter(proj => !isRemoteProject(proj))
-      : [activeProject];
+  const organizationProjects = allProjects.filter(proj => !isRemoteProject(proj));
 
   const projects = sortProjects(organizationProjects);
```
Comment (jackkav) [inline packages/insomnia/src/ui/routes/workspace.tsx]: can clean this up, use db.find

File: packages/insomnia/src/ui/components/present-users.tsx (hunk 26 above)
Comment (jackkav) [reply packages/insomnia/src/ui/components/present-users.tsx]: done in https://github.com/Kong/insomnia/pull/6552

**Review by marckong — APPROVED** [review]
